### PR TITLE
Toolbar async prototype

### PR DESCRIPTION
**Pull request by nces**

### CODE REVIEW

**Comment by nces** [comment]
Possible toolbar usage with async wrappers.

File: Toolbar1Async/manifest.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (nces) [inline Toolbar1Async/manifest.json]: A couple things to note:
- The `PluginEntry` defines the "entry point" of the plugin, effectively the "index.html" for any website.
- There is no `toolbar.json`. This additional file seems unnecessary to me. If it's really desirable, I can make it work for the async case.
- Currently you can't make a panel and a toolbar in a single plugin, but that is completely doable, I just didn't workout what the manifest would look like in that case. e.g., it seems reasonable to be able to have a single "Plugin" that creates several panels and toolbar buttons, all driven off the same HTML page.
- It seems like in pretty much every use case, paths referred to in the manifest will refer to files relative to the plugin directory, which makes the use of `PLUGINLOCATION` superfluous, for relative paths at least. See comments in the corresponding FormIt360 PR.

**Comment by nces** [comment]
As with the original async wrapper change, this does not effect existing plugins.

The main motivation here (for me at least) is the single execution context for the user (eliminating the need for the `Scripts` manifest entry). See the comment in Toolbar1Async/manifest.json for some additional changes I made (because they were bothering me...).

File: Toolbar1Async/index.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        window.AlertFormItVersion = async (data) => {
+            const version = await FormIt.GetVersion();
+            alert(`Toolbar clicked! Event data = ${JSON.stringify(version)}`);
+        };
+
+        window.CreateBlock = async (data) => {
+            const pt1 = WSM.Geom.Point3d(-6, -6, 0);
+            const pt2 = WSM.Geom.Point3d(6, 6, 12);
+            const history = FormIt.GroupEdit.GetEditingHistoryID();
+            WSM.APICreateBlock(await history, await pt1, await pt2);
```
Comment (jhauswirth) [inline Toolbar1Async/index.html]: For each toolbar button you used to be able to assign a shortcut to the action.  How do you do that now?

File: Toolbar1Async/manifest.json (hunk 1 above)
Comment (jhauswirth) [reply Toolbar1Async/manifest.json]: I think being able to handle a plugin that has both a panel and toolbar would be good to have.  How much work would that be?

**Review by mattributes — APPROVED** [review]

**Review by sum1991 — APPROVED** [review]
LGTM

**Review by jhauswirth — APPROVED** [review]

**Comment by nces** [comment]
I tried making some minor tweaks to the manifest and it looks like toolbars and panels can pretty easily coexist in a single plugin. I changed the example I added to showcase using shared code for toolbar buttons and a panel.

File: Toolbar1Async/index.html (hunk 2 above)
Comment (nces) [reply Toolbar1Async/index.html]: Good question. That may not be possible. How did that work before?

File: Toolbar1Async/manifest.json (hunk 1 above)
Comment (nces) [reply Toolbar1Async/manifest.json]: Looks like it already works (more or less).

File: Toolbar1Async/index.html (hunk 2 above)
Comment (jhauswirth) [reply Toolbar1Async/index.html]: FormIt.Commands.RegisterJSCommand("ToolbarPlugin.Button");
And in the json for the toolbar-
https://github.com/FormIt3D/FormItExamplePlugins/blob/master/Toolbar1/toolbar.json#L6

File: Toolbar1Async/manifest.json (hunk 1 above)
Comment (josh-goldstein-adsk) [reply Toolbar1Async/manifest.json]: It's nice that a single plugin can be a panel and a toolbar. Nice work!

**Pull request merged**